src/olympia/search/filters.py

```diff
@@ -272,9 +272,20 @@ def primary_should_rules(self, search_query, analyzer):
 
         # Apply rules to search on few base fields. Some might not be present
         # in every document type / indexes.
-        for k, v in rules:
+        for query_cls, opts in rules:
             for field in ('name', 'slug', 'listed_authors.name'):
-                should.append(k(**{field: v}))
+                should.append(query_cls(**{field: opts}))
+
+        # Exact matches need to be queried against a non-analyzed field. Let's
+        # do a term query on `name.raw` for an exact match against the add-on
+        # name and boost it since this is likely what the user wants.
+        # Use a super-high boost to avoid `description` or `summary`
+        # getting in our way.
+        should.append(query.Term(**{
+            'name.raw': {
+                'value': search_query, 'boost': 100
```

Comment:
Might be a lot, but I suppose we can tweak that and lower it if necessary.

Reply:
Yeah, unfortunately, the scoring for `summary` and `description` adds up over time and trumps the exact match relatively quickly. I chose a huge boost for exact matches to get this going and work on debugging why summary + description have such a huge impact (even though the boost numbers are theoretically negative) but that might take a while.